sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -228,17 +228,24 @@ async def __anext__(self):
             except _ResponseStopIteration:
                 self.response.internal_response.close()
                 raise StopAsyncIteration()
-            except (ChunkedEncodingError, ConnectionError):
+            except (ChunkedEncodingError, ConnectionError) as ex:
                 retry_total -= 1
                 if retry_total <= 0:
                     retry_active = False
                 else:
                     await asyncio.sleep(retry_interval)
                     headers = {'range': 'bytes=' + str(self.downloaded) + '-'}
-                    resp = await self.pipeline.run(self.request, stream=True, headers=headers)
-                    if resp.http_response.status_code == 416:
-                        raise
-                    chunk = await self.response.internal_response.content.read(self.block_size)
+                    try:
+                        resp = await self.pipeline.run(self.request, stream=True, headers=headers)
+                        if not resp.http_response:
+                            raise
+                        if resp.http_response.status_code == 416:
+                            raise
+                        chunk = await self.response.internal_response.content.read(self.block_size)
```

Comment:
This line re-uses the original response that just raised an exception in L223 (it does *not* use the new response created on l. 239!). But: what is the point to re-try a read after a `ConnectionError`? How can that ever succeed? Even if for `ConnectionError`, there might be cases it makes sense, it's clearly wrong for a `ChunkedEncodingError`: this just means there is a serious error in the data stream parsing. Re-using the data stream that just observed such a parsing error is wrong.